_tutorials/development.md

```diff
@@ -22,32 +25,67 @@ $ ssh <NETID>@{{site.data.constants.semester }}-{{ site.data.constants.departmen
+
+An alternative to using the UIUC VPN is to SSH twice. You can first SSH into your EWS account and then into your personal VM. Just remember that this causes potentially double the network lag!
+
+Additionally, make sure your VM is on and running, which can be done [at the VM dashboard](https://vc.cs.illinois.edu/). Use [this](https://csid-basic-apps.cs.illinois.edu/) to lookup your VM name if needed.  
+**Note: Both of the above links also require campus wifi or a VPN**
+
+
+## Setting Up Git
+
+You will be using `git` to submit all your assignments in this course. First go to [the repository creator](https://edu.cs.illinois.edu/create-gh-repo/{{site.data.constants.semester}}_{{site.data.constants.department_code}}{{site.data.constants.course_number}}/).
 
-### Enabling password-less authentication with your VM
+### Configuring Git
+
+**Once you are in your VM**, you'll need to set up some global defaults. The name and email you input here will be used to mark your commits.
 
-If you have generated an SSH key on your local machine (see [Using SSH keys](#using-ssh-keys) below), you can export your public key to your VM using the following command:
 ```console
-$ ssh-copy-id <NETID>@{{site.data.constants.semester }}-{{ site.data.constants.department_code }}{{ site.data.constants.course_number }}-<NUM>.cs.illinois.edu
+git config --global user.name "FIRST_NAME LAST_NAME"
+git config --global user.email NETID@example.com
 ```
 
-## Authenticating with GitHub
+Make sure to replace `FIRST_NAME` and `LAST_NAME` and `NETID@example.com` with your information. For example:
+
+```console
+git config --global user.name "Lawrence Angrave"
+git config --global user.email "angrave@illinois.edu"
+```
 
-There are two ways that we recommend for authenticating with GitHub
+### Cloning your course repository
 
-### Using SSH keys
+Checkout your repository as follows:
+
+```console
+git clone https://github.com/illinois-cs-coursework/{{site.data.constants.semester }}_{{ site.data.constants.department_code }}{{site.data.constants.course_number}}_NETID.git
+```
+
+Unfortunately, cloning a repository is not so easy. If you follow git's prompt and submit your username and password, you will likely see an error message like the following. If so, continue to the instructions in [Authenticating with GitHub](#authenticating-with-github).
+
+```console
+remote: Support for password authentication was removed on August 13, 2021.
+remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
+fatal: Authentication failed for 'https://github.com/illinois-cs-coursework/{{site.data.constants.semester }}_{{ site.data.constants.department_code }}{{site.data.constants.course_number}}_NETID.git'
+```
+
+### Authenticating with GitHub
+
+There are three ways that we recommend for authenticating with GitHub
+
+#### Option 1: Using SSH keys
```

Comment:
Maybe hyperlink "SSH keys" to the official Github article on how to use ssh keys 
[](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)

Reply:
Updated